framework/helpers/BaseStringHelper.php

```diff
@@ -104,12 +104,15 @@ public static function dirname($path)
      */
     public static function truncate($string, $length, $suffix = '...', $encoding = null, $asHtml = false)
     {
+        if ($encoding === null) {
+            $encoding = Yii::$app ? Yii::$app->charset : 'UTF-8';
```

Comment:
> Standalone valdator and helper running will become impossible in case you need default encoding different from UTF-8.

For the helper it is no problem as long as you specify the `$encoding` param of the method.
For the validator we could add a property that defaults to `Yii::$app->charset` or 'UTF-8' if not set.